File: 1306-minimum-absolute-difference/minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> list=new ArrayList<>();
        Arrays.sort(arr);
        int min=Integer.MAX_VALUE;
        for(int i=1;i<arr.length;i++){
            min = Math.min(min,Math.abs(arr[i]-arr[i-1]));
        }
        for(int i=1;i<arr.length;i++){
            if(min==Math.abs(arr[i]-arr[i-1]))
            list.add(Arrays.asList(arr[i-1],arr[i]));
        }
        return list;

    }
}

File: 1306-minimum-absolute-difference/minimum-absolute-difference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a,b)=>a-b);
    let res=[];
    let min=Infinity;
    for(let i=1;i<arr.length;i++){
        min=Math.min(min,arr[i]-arr[i-1]);
    }
    for(let i=1;i<arr.length;i++){
        if(Math.abs(arr[i-1]-arr[i])===min){
            res.push([arr[i-1],arr[i]]);
        }
    }
    return res;
};